### ISSUE TITLE
[TRACE-X]⭐🚀  Switch to IRS Order API for scalable Job Processing

### ISSUE THREAD

**Issue by mkanal** [eclipse-tractusx/sig-release#946]
# Overview

## Explain the topic in 2 sentences
Trace-X currently utilizes a Job API that processes each job individually, resulting in limited scalability.
By integrating the IRS Order API, irs processes multiple jobs in batches, allowing for parallel processing of twins, thereby improving overall system efficiency and scalability.

## What's the benefit?
Usage of IRS Order API which is designed to process big amount of data and focussing more on performance. 

## What are the Risks/Dependencies ?
- Nothing to report 

## Detailed explanation
Trace-X to switch from the current Job API to the IRS Order API,

### Current implementation
Trace-X used current Job API 

### Proposed improvements
Trace-X used IRS Order API

### Contributor

- Cofinity-X  @rogocof 
- Cofinity-X @mkanal 

### Committer

- @ds-jhartmann 
- @ds-mwesener 
- @ds-lcapellino 

----

## Current State:

Trace-X currently utilizes a Job API that processes each job individually, resulting in limited scalability.

By integrating the IRS Order API, irs processes multiple jobs in batches, allowing for parallel processing of twins, thereby improving overall system efficiency and scalability.

## User Story

As a customer 
I want Trace-X to switch from the current Job API to the IRS Order API,
so that we can achieve scalable job processing by enabling batch processing of multiple twins in parallel. 

## Hint 

````
{
  ..
  "jobTimeout": 3600,
 ...
  "timeout": 43200
}
````

## Acceptance Criteria 

- [x] Trace-X uses IRS Order API for retrieving assets. 
- [x] Trace-X handles batches results by irs in a sufficient way and handles error situations sufficiently.
- [x]  Trace-X handles and monitors order status to ensure that all transferred twins have been processed 
- [x]  Trace-X tries to cancel orders if defined thresholds of 12 h is reached. PUT /irs/order/{id}
- [x]  Performance metrics should demonstrate improved job processing times compared to the existing individual job processing approach.
- [x] Ensure that the new batch processing in Trace-X is reliable and handles errors gracefully.
- [x]  Execute and document a benchmark test after implementation in Confluence 

----

## Test Cases

Feature: Job Processing via IRS Order API in Trace-X

  As a system integrator,
  I want to ensure that job processing in Trace-X continues to function correctly
  after switching to the IRS Order API,
  so that existing functionality is not disrupted.

  Background:
    Given the Trace-X system is configured to use the IRS Order API for job processing
    And valid jobs (globalAssetIds/aasId and BPNL Tuples) are available

  Scenario: Successful processing of a job using IRS Order API
    When the job is submitted to Trace-X
    Then the job is accepted by the system
    And the job is processed without errors
    And the job result is available in the expected output location
    And the job status is marked as "COMPLETED"

  # Note: Improved performance and scalability are measured via separate KPIs and are not part of this feature test.

## Architectural Relevance

The following items are ensured (answer: yes) after this issue is implemented:

- [x] This feature aligns with our current architectural guidelines
  - **Data Sovereignty:** All data sharing activities across company boundaries follow the [Catena-X Regulatory Framework](https://catenax-ev.github.io/docs/next/regulatory-framework/governance-framework), in particular the [Data Exchange Governance](https://catenax-ev.github.io/docs/next/regulatory-framework/20000ft/data-exchange-governance), and the [Dataspace Protocol](https://docs.internationaldataspaces.org/dataspace-protocol/overview/readme) via a compliant Connector (like the [tractusx-edc](https://github.com/eclipse-tractusx/tractusx-edc) or similar, see [Connector KIT](https://eclipse-tractusx.github.io/docs-kits/next/category/connector-kit))
    - [x] [CX-0010 Business Partner Number](https://catenax-ev.github.io/docs/next/standards/CX-0010-BusinessPartnerNumber)
    - [x] [CX-0013 Identity of Member Companies](https://catenax-ev.github.io/docs/next/standards/CX-0013-IdentityOfMemberCompanies)
    - [x] [CX-0018 Data Space Connectivity (EDC)](https://catenax-ev.github.io/docs/next/standards/CX-0018-DataspaceConnectivity)
    - [x] [CX-0049 DID Document Schema](https://catenax-ev.github.io/docs/next/standards/CX-0049-DIDDocumentSchema)
    - [x] [CX-0050 Framework Agreement Credential](https://catenax-ev.github.io/docs/next/standards/CX-0050-FrameworkAgreementCredential)
    - [x] [CX-0149 Verified Company Identity](https://catenax-ev.github.io/docs/next/standards/CX-0149-Dataspaceidentityandidentification)
  - **Interoperability:** Digital Twins are used (compliant to the [Digital Twin KIT](https://eclipse-tractusx.github.io/docs-kits/next/category/digital-twin-kit) and the [Industry Core KIT](https://eclipse-tractusx.github.io/docs-kits/next/category/industry-core-kit))
    - [x] [CX-0001 EDC Discovery API](https://catenax-ev.github.io/docs/next/standards/CX-0001-EDCDiscoveryAPI)
    - [x] [CX-0002 Digital Twins in Catena-X](https://catenax-ev.github.io/docs/next/standards/CX-0002-DigitalTwinsInCatenaX)
    - [x] [CX-0018 Data Space Connectivity (EDC)](https://catenax-ev.github.io/docs/next/standards/CX-0018-DataspaceConnectivity)
  - **Data Format:**
    - [x] The data model is based on a [published Semantic Model](https://github.com/eclipse-tractusx/sldt-semantic-models)
- [x] The impact on the overall system architecture has been assessed. The Feature does not require changes to the architecture or any existing standard? Please have a look here on the [overarching architecture](https://eclipse-tractusx.github.io/docs/tutorials/e2e/inform/architecture)
- [x] Potential risks or conflicts with existing architecture has been assessed

**Justification:** _(Fill this out, if at least one of the checkboxes above cannot be ticked. Contact the Architecture Management Committee to get an approval for the justification)_

## Additional information

- [x] I am aware that my request may not be developed if no developer can be found for it. I'll try to contribute a developer (bring your own developer)

**Comment by stephanbcbauer** [comment]
Some hints from Release Management (@ther3sa) and Tractus-X Project Lead (@stephanbcbauer)
- **Status currently in Backlog** -> Since for Backlog some requirements are missing ... please update them
- Please add missing sections from the feature template, or fill them out
- Please add assignee (overall responsible person who drives the feature)

**Comment by ds-mwesener** [comment]
Hi @ther3sa we have included the story in the current release and it was part of our testing. Therefore I close the ticket as done. Please let me know if you have any doubts.
